app/cdap/components/DataPrep/Directives/ChangeDataType/DecimalOptions.tsx

```diff
@@ -0,0 +1,293 @@
+import styled from 'styled-components';
+import T from 'i18n-react';
+
+import {
+  ISubmenuProps,
+  SubmenuContainer,
+} from 'components/DataPrep/Directives/ChangeDataType/submenu';
+
+import { makeStyles } from '@material-ui/core/styles';
+import Paper from '@material-ui/core/Paper';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import HelpIcon from '@material-ui/icons/Help';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Button from '@material-ui/core/Button';
+import Tooltip from '@material-ui/core/Tooltip';
+
+const PREFIX = 'features.DataPrep.Directives.ChangeDataType.decimalConfig';
+const ROUNDING_PREFIX = `${PREFIX}.roundingOptions`;
+
+const ID_PREFIX = 'DataPrep-Directives-ChangeDataType-decimal';
+
+const SCALE_INPUT_ID = `${ID_PREFIX}-scaleInputId`;
+const ROUNDING_INPUT_ID = `${ID_PREFIX}-roundingInputId`;
+const ROUNDING_LABEL_ID = `${ID_PREFIX}-roundingLableId`;
+
+interface IRoundingMode {
+  value: string;
+  text: string;
+  description?: string;
+}
+
+const ROUNDING_MODES: IRoundingMode[] = [
+  {
+    value: 'CEILING',
+    text: T.translate(`${ROUNDING_PREFIX}.CEILING.label`).toString(),
+    description: T.translate(`${ROUNDING_PREFIX}.CEILING.description`).toString(),
+  },
+  {
+    value: 'DOWN',
+    text: T.translate(`${ROUNDING_PREFIX}.DOWN.label`).toString(),
+    description: T.translate(`${ROUNDING_PREFIX}.DOWN.description`).toString(),
+  },
+  {
+    value: 'FLOOR',
+    text: T.translate(`${ROUNDING_PREFIX}.FLOOR.label`).toString(),
+    description: T.translate(`${ROUNDING_PREFIX}.FLOOR.description`).toString(),
+  },
+  {
+    value: 'HALF_DOWN',
+    text: T.translate(`${ROUNDING_PREFIX}.HALF_DOWN.label`).toString(),
+    description: T.translate(`${ROUNDING_PREFIX}.HALF_DOWN.description`).toString(),
+  },
+  {
+    value: 'HALF_EVEN',
+    text: T.translate(`${ROUNDING_PREFIX}.HALF_EVEN.label`).toString(),
+    description: T.translate(`${ROUNDING_PREFIX}.HALF_EVEN.description`).toString(),
+  },
+  {
+    value: 'HALF_UP',
+    text: T.translate(`${ROUNDING_PREFIX}.HALF_UP.label`).toString(),
+    description: T.translate(`${ROUNDING_PREFIX}.HALF_UP.description`).toString(),
+  },
+  {
+    value: 'UNNECESSARY',
+    text: T.translate(`${ROUNDING_PREFIX}.UNNECESSARY.label`).toString(),
+    description: T.translate(`${ROUNDING_PREFIX}.UNNECESSARY.description`).toString(),
+  },
+  {
+    value: 'UP',
+    text: T.translate(`${ROUNDING_PREFIX}.UP.label`).toString(),
+    description: T.translate(`${ROUNDING_PREFIX}.UP.description`).toString(),
+  },
+];
+
+const useStyles = makeStyles((theme) => ({
```

Comment:
could you do this with styled-components?

Reply:
Nope. I tried it. Seems the only way to customise the mui tooltip is to use the classes prop.